parsers/rebase/rebase.go

```diff
@@ -0,0 +1,264 @@
+                               (4) = N4-methylcytosine
+
+                If the methylation information is different for the 3' strand,
+                X2 and Y2 are given as well.
+
+<MICROORGANISM> Organism from which this enzyme had been isolated.
+<SOURCE>        Either an individual or a National Culture Collection.
+<COMMERCIAL AVAILABILITY>
+                Each commercial source of restriction enzymes and/or methylases
+                listed in REBASE is assigned a single character abbreviation
+                code.  For example:
+
+                K        Takara (1/98)
+                M        Boehringer Mannheim (10/97)
+                N        New England Biolabs (4/98)
+
+                The date in parentheses indicates the most recent update of
+                that organization's listings in REBASE.
+
+<REFERENCES>only the primary references for the isolation and/or purification
+of the restriction enzyme or methylase, the determination of the recognition
+sequence and cleavage site or the methylation specificity are given.
+
+
+REBASE codes for commercial sources of enzymes
+
+                B        Life Technologies (3/21)
+                C        Minotech Biotechnology (3/21)
+                E        Agilent Technologies (8/20)
+                I        SibEnzyme Ltd. (3/21)
+                J        Nippon Gene Co., Ltd. (3/21)
+                K        Takara Bio Inc. (6/18)
+                M        Roche Applied Science (4/18)
+                N        New England Biolabs (3/21)
+                O        Toyobo Biochemicals (8/14)
+                Q        Molecular Biology Resources - CHIMERx (3/21)
+                R        Promega Corporation (11/20)
+                S        Sigma Chemical Corporation (3/21)
+                V        Vivantis Technologies (1/18)
+                X        EURx Ltd. (1/21)
+                Y        SinaClon BioScience Co. (1/18)
+
+<1>AaaI
+<2>XmaIII,BseX3I,BsoDI,BstZI,EagI,EclXI,Eco52I,SenPT16I,TauII,Tsp504I
+<3>C^GGCCG
+<4>
+<5>Acetobacter aceti ss aceti
+<6>M. Fukaya
+<7>
+<8>Tagami, H., Tayama, K., Tohyama, T., Fukaya, M., Okumura, H., Kawamura, Y., Horinouchi, S., Beppu, T., (1988) FEMS Microbiol. Lett., vol. 56, pp. 161-166.
+
+```
+******************************************************************************/
+
+// Enzyme represents a single enzyme within the Rebase database
+type Enzyme struct {
+	Name                   string   `json:"name"`
+	Isoschizomers          []string `json:"isoschizomers"`
+	RecognitionSequence    string   `json:"recognitionSequence"`
+	MethylationSite        string   `json:"methylationSite"`
+	MicroOrganism          string   `json:"microorganism"`
+	Source                 string   `json:"source"`
+	CommercialAvailability []string `json:"commercialAvailability"`
+	References             string   `json:"references"`
+}
+
+// ParseRebase parses the Rebase database into a map of enzymes
+func ParseRebase(file []byte) map[string]Enzyme {
```

Comment:
Should this be exposed to end users? It seems like this would be better as a private function for Poly developers to use unless NEB asks us to export it for them to use.

Reply:
I was thinking of having it exposed for end users in case they want to store the enzymes elsewhere, in which they can just pass bytes in to the function. Kinda similar to the idea of `GetEnzymeMap`, where you actually just directly download the file and pass it into ParseRebase.